avocado/core/test.py

```diff
@@ -783,3 +783,17 @@ class ReplaySkipTest(SkipTest):
     """
 
     _skip_reason = "Test skipped due to a job replay filter!"
+
+
+class TestError(Test):
+    """
+    Generic test error.
+    """
+
+    def __init__(self, *args, **kwargs):
+        exception = kwargs.pop('exception')
+        Test.__init__(self, *args, **kwargs)
+        self.exception = exception
+
+    def unknown(self):
```

Comment:
Oh, I didn't noticed that would require this modification. The `TestError` should define `test` instead of `unknown`.

Anyway I tested it with the `multiple_tests` test and as the static analysis is used, the exception contains the correct module, names and everything.

So I'm sorry for tricking you into this, but it should stay as it was in the v3 (`def test`) and discover it as `test`. It provides the right message.